src/cascadia/TerminalApp/TmuxControl.cpp

```diff
@@ -0,0 +1,1480 @@
+                _event.type = ATTACH;
+            }
+            else
+            {
+                _event.type = RESPONSE;
+            }
+        }
+        else if (std::regex_match(line, REG_ERROR))
+        {
+            // Remove the extra '\n' we added
+            _Print(std::wstring(_event.response.begin(), _event.response.end() - 1));
+            _event.response.clear();
+            _event.type = NOTHING;
+        }
+
+        // tmux specific rules
+        else if (std::regex_match(line, REG_DETACH))
+        {
+            _event.type = DETACH;
+        }
+        else if (std::regex_match(line, matches, REG_LAYOUT_CHANGED))
+        {
+            _event.windowId = std::stoi(matches.str(1));
+            _event.type = LAYOUT_CHANGED;
+        }
+        else if (std::regex_match(line, matches, REG_OUTPUT))
+        {
+            _event.paneId = std::stoi(matches.str(1));
+            _event.response = matches.str(2);
+            _event.type = OUTPUT;
+        }
+        else if (std::regex_match(line, matches, REG_SESSION_CHANGED))
+        {
+            _event.type = SESSION_CHANGED;
+            _event.sessionId = std::stoi(matches.str(1));
+        }
+        else if (std::regex_match(line, matches, REG_WINDOW_ADD))
+        {
+            _event.windowId = std::stoi(matches.str(1));
+            _event.type = WINDOW_ADD;
+        }
+        else if (std::regex_match(line, matches, REG_WINDOW_CLOSE))
+        {
+            _event.type = WINDOW_CLOSE;
+            _event.windowId = std::stoi(matches.str(1));
+        }
+        else if (std::regex_match(line, matches, REG_WINDOW_PANE_CHANGED))
+        {
+            _event.type = WINDOW_PANE_CHANGED;
+            _event.windowId = std::stoi(matches.str(1));
+            _event.paneId = std::stoi(matches.str(2));
+        }
+        else if (std::regex_match(line, matches, REG_WINDOW_RENAMED))
+        {
+            _event.windowId = std::stoi(matches.str(1));
+            _event.response = matches.str(2);
+            _event.type = WINDOW_RENAMED;
+        }
+        else if (std::regex_match(line, matches, REG_UNLINKED_WINDOW_CLOSE))
+        {
+            _event.type = UNLINKED_WINDOW_CLOSE;
+            _event.windowId = std::stoi(matches.str(1));
+        }
+        else
+        {
+            if (_event.type == BEGIN)
+            {
+                _event.response += line + L'\n';
+            }
+            else
+            {
+                // Other events that we don't care, do nothing
+                _event.type = NOTHING;
+            }
+        }
+
+        if (_event.type != BEGIN && _event.type != NOTHING)
+        {
+            auto& e = _event;
+            _dispatcherQueue.TryEnqueue([this, e]() {
+                _EventHandler(e);
+            });
+            _event.response.clear();
+        }
+
+        return;
+    }
+
+    // From tmux to controller through the dcs. parse it per line.
+    bool TmuxControl::_Advance(wchar_t ch)
```

Comment:
I've written #19640 to improve the performance of this PR. I've already modified this function in my branch to process entire `std::wstring_view`s at a time.